facter/facter.go

```diff
@@ -197,11 +191,52 @@ func (f *Facter) GetConfigPolicy() (*cpolicy.ConfigPolicy, error) {
 
 // ------------ helper functions --------------
 
+// parseFacts parses facts returned by facter and extracts metricTypes from it
+func parseFacts(f facts) ([]plugin.MetricType, error) {
+
+	var metricTypes []plugin.MetricType
+
+	// create types with given namespace
+	for name, value := range f {
+		if val, ok := value.(map[string]interface{}); ok {
+			ms, err := createMetricsSubArray(name, val)
+			if err != nil {
+				return nil, err
+			}
+			metricTypes = append(metricTypes, ms...)
+		} else {
+			metricType := *plugin.NewMetricType(createNamespace(name), time.Now(), nil, "", value)
+			metricTypes = append(metricTypes, metricType)
+		}
+	}
+
+	return metricTypes, nil
+}
+
+// createMetricsSubArray creates an array of metrics for map values returned by facter
+func createMetricsSubArray(name string, value map[string]interface{}) ([]plugin.MetricType, error) {
+
+	var ms []plugin.MetricType
+
+	namespaces := []string{}
+
+	err := utils.FromMap(value, name, &namespaces)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, namespace := range namespaces {
+		ns := strings.Split(namespace, "/")
+		val := utils.GetValueByNamespace(value, ns[1:])
+		m := *plugin.NewMetricType(createNamespace(ns...), time.Now(), nil, "", val)
+		ms = append(ms, m)
+	}
+
+	return ms, nil
+}
+
 // validateNamespace checks namespace intel(vendor)/facter(prefix)/FACTNAME
 func validateNamespace(namespace core.Namespace) error {
```

Comment:
Perhaps this functionis unnecessary and namespace routing is carried by framework but if you decided to leave it here the check should be changed to `len(namespace) >= 3`.